### PR TITLE
Load global vars when saving app objects

### LINKED ISSUE #295: Error when trying to initialize AppDriver from inside a function
I want to have a way to easily create an AppDriver for different shiny modules. I provide the UI and the server of a module, and a shiny app is created with that module, then an AppDriver is created for this shiny app. I get an error when doing that. Full details below.

Here's an example little module to test:

```r
library(shiny)

my_mod_ui <- function(id) {
  ns <- NS(id)
  tagList(
    textInput(ns("name"), "name", "test"),
    textOutput(ns("greet"))
  )
}

my_mod_server <- function(id) {
  moduleServer(id, function(input, output, session) {
    output$greet <- renderText(input$name)
  })
}
```

And here is a function for easily wrapping a given module in a shiny app, a way to "graduate" a module to a real app:

```r
make_app <- function(mod_ui, mod_server) {
  shiny::shinyApp(
    shiny::fluidPage(mod_ui),
    function(input, output, session) mod_server
  )
}
```

Now I can create a shiny app easily from a module, for example:

```r
app <- make_app(my_mod_ui("test"), my_mod_server("test"))
```

You can run this `app` and it works.

To create an AppDriver for this, I can do:

```r
make_driver <- function(app) {
  shinytest2::AppDriver$new(app)
}
make_driver(app)
```

And this works. However, my actual goal is to have just one function where I provide the module, and it creates the AppDriver. So if I take the same code and put it into one function, it looks like this:

```r
make_driver <- function(mod_ui, mod_server) {
  app <- make_app(mod_ui, mod_server)
  shinytest2::AppDriver$new(app)
}
```

But this doesn't work. If I call `make_driver(my_mod_ui("test"), my_mod_server("test"))`, I get an error:

```
Error in `app_initialize()`:
! Identified global objects via static code inspection (function (input, output, session); mod_server). session must be a ShinySession or session_proxy object.


ℹ You can inspect the failed AppDriver object via `rlang::last_error()$app`
ℹ AppDriver logs:
{shinytest2} R info 17:20:51.65 Start AppDriver initialization
{shinytest2} R info 17:20:51.67 Error while initializing AppDriver:
                                Identified global objects via static code inspection (function (input, output, session); mod_server). session must be a ShinySession or session_proxy object.

Caused by error in `callModule()`:
! Identified global objects via static code inspection (function (input, output, session); mod_server). session must be a ShinySession or session_proxy object.
Run `rlang::last_error()` to see where the error occurred.
```

### LINKED ISSUE #303: How to use a variable inside a shiny app?
I'm trying to create a unit test where I set a variable at the start of the {testthat} unit test and use that variable in a shiny app, then use {shinytest2} to test the app. However, it seems like I can't refer to that variable when wrapping my app in an AppDriver.

Example:

```r
foo <- 5

app <- shiny::shinyApp(
  ui <- shiny::fluidPage(),
  server = function(input, output, session) {
    foo
  }
)

driver <- shinytest2::AppDriver$new(app)
driver$get_logs()
```

You'll see the error message `object 'foo' not found`

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #303 
Fixes #295

- [x] Try using global env when assigning
	* ~~Ans: Shimming an environment between the server and global env~~
	* Ans: Using global env. Performing multi asign using `list2env(envir=globalenv())`
- [x] Try using `{future}` / callr to pass app object instead of saving to temp arg
	* Ans: While `{future}` does a great job at capturing what is needed, it does not work well with stdout/stderr. This is a deal breaker.

### CODE REVIEW

File: inst/internal/app-template.R [hunk 1]
```diff
@@ -6,5 +6,9 @@ lapply(`_data`$packages, library, character.only = TRUE)
 for (prefix in names(`_data`$resources)) {
   shiny::addResourcePath(prefix, `_data`$resources[[prefix]])
 }
+for (name in names(`_data`$globals)) {
+  message("setting global name: ", name)
+  assign(name, `_data`$globals[[name]], environment(`_data`$server))
```
Comment (jcheng5) [inline inst/internal/app-template.R:11]: Consider using `globalenv()` as the third argument

**Comment by jcheng5** [comment]
Am I correct that the main three cases we care about are:

1. Writing tests for a nontrivial app that is in an app.R
2. Writing tests for a nontrivial app that is using golem/rhino(?)
3. Writing tests for trivial apps that exercise some feature or component that is not specific to a single app

The kind of cross-process de/serialization that this PR is fixing, totally makes sense to me for case 3. It makes me a lot less comfortable if people are relying on it for case 2. I'd love to see us make a formal recommendation for using shinytest2 with golem/rhino that doesn't involve serializing any code objects.

**Comment by daattali** [comment]
The biggest benefit that I personally wanted from this is to be able to test a shiny module.

```r
library(shiny)

mod_ui <- function(id) {
  ns <- NS(id)
  textOutput(ns("foo"))
}

mod_server <- function(id) {
  moduleServer(
    id,
    function(input, output, session) {
      output$foo <- renderText("bar")
    }
  )
}

app <- shinyApp(
  mod_ui("test"), 
  function(input, output, session) { mod_server("test") }
)

driver <- shinytest2::AppDriver$new(app)
driver$get_logs()  #  Error in mod_server("test") : could not find function "mod_server"
```

**Comment by daattali** [comment]
This PR does fix my usecase in #303

**Pull request merged**